src/test/analyzer_test.ts

```diff
@@ -47,16 +48,155 @@ suite('Analyzer', () => {
+              Array.from(document.getFeatures(false))
+                  .filter(
+                      (d) => d instanceof Document && d.isInline) as Document[];
+          assert.equal(inlineDocuments.length, 1);
+          const inlineJsDocument = inlineDocuments[0];
+
+          // The inline document can find the container's imported features
+          const subBehavior =
+              inlineJsDocument.getOnlyAtId('behavior', 'TestBehavior');
+          assert.equal(subBehavior.className, 'TestBehavior');
+        });
+
+    test(
+        'an external script can find features from its container document',
+        async() => {
+          const document = await analyzer.analyze(
+              'static/script-tags/external/test-element.html');
+
+          const htmlScriptTags = Array.from(document.getByKind('html-script'));
+          assert.equal(htmlScriptTags.length, 1);
+
+          const htmlScriptTag = htmlScriptTags[0] as HtmlScriptTagImport;
+          const scriptDocument = htmlScriptTag.document;
+
+          // The inline document can find the container's imported features
+          const subBehavior =
+              scriptDocument.getOnlyAtId('behavior', 'TestBehavior');
+          assert.equal(subBehavior.className, 'TestBehavior');
+        });
+
+
+    // This test is nearly identical to the previous, but covers a different
+    // issue.
+    // The Polymer element-descriptor must find behaviors while resolving,
+    // and if inline document's don't add a document feature for their container
+    // until after resolution, then the element can't find them and throws.
+    test(
+        'an inline document can find features from its container document',
+        async() => {
+          const document = await analyzer.analyze(
+              'static/analysis/behaviors/elementdir/element.html');
+
+          // TODO(justinfagnani): make a shallow option and check that this only
+          // has
+          // itself and an inline document, but not the sub-document. For now
+          // check
+          // that this fixture has 6 documents: element.html, behavior.html,
+          // subbehavior.html, and their inline js documents
+          const documents = document.getByKind('document');
+          assert.equal(documents.size, 6);
+
+          const inlineDocuments =
+              Array.from(document.getFeatures(false))
+                  .filter(
+                      (d) => d instanceof Document && d.isInline) as Document[];
+          assert.equal(inlineDocuments.length, 1);
+
+          // This is the main purpose of the test: get a feature from the inline
+          // document that's imported by the container document
+          const behaviorJsDocument = inlineDocuments[0];
+          const subBehavior = behaviorJsDocument.getOnlyAtId(
+              'behavior', 'MyNamespace.SubBehavior');
+          assert.equal(subBehavior.className, 'MyNamespace.SubBehavior');
+        });
+
```

Comment:
I agree that it's tricky, but I think supporting it one way or another is needed. Design TBD, but I think a TODO documenting the intention would be nice here.